### ISSUE TITLE
Allow the specification of a different sources (work) directory

### ISSUE THREAD

**Issue by tadeu** [conda/conda-build#720]
Currently we can only specify the conda-build `root-dir` in `.condarc`, but on Windows it must be a path in the same drive where `Miniconda` is installed.

It would be interesting to allow the specification of a `work-dir-base` where the sources would be checked out under `work-dir-base/platform/pkg-name`. This would be used both for `conda build` and `conda build --source`.

For example, it could be specified that the base is `S:\` (a subst drive) or `/usr/local/src/` on Linux. In this case, Python 2.7 would be built on `S:\win-64\python-2.7.10` while Boost would be built on `S:\win64\boost-1.60.0`.

This is interesting because usually the sources paths are hardcoded in the debug information (it can be "relocated" during a debug session, but this is annoying). Allowing a common place where sources are built would improve both debugging sessions and source files organization (more than one source at a time).

Also, an advantage of putting the platform in the PATH is that this enables easy parallelization of builds (i.e., build win-32 and win-64 at the same time).

**Comment by tadeu** [comment]
Just need an ok and I can work on a PR for this :)

**Comment by msarahan** [comment]
I think this is a good idea.  Sometimes it is very hard to debug dependencies with the limit of only 1 package "in progress" at once.  Parallelization sounds nice, but may cause more trouble than anticipated (oversubscription could happen pretty easily.)  Many builds are already set up to use as many cores as the system has, and besides that, these things often get IO-bound anyway.  Still - nice to have the possibility of parallel, but perhaps not the best default.

One thing to consider is to make certain that builds can be kept "clean" - if this PR makes it easier to accidentally include uncommitted custom code by changing one source tree and including those changes in the other package, then that's not good.

Just some thoughts here, not an official "OK."  Pinging @ilanschnell @kalefranz @csoja - one of them should comment here for your "OK"

**Comment by tadeu** [comment]
> One thing to consider is to make certain that builds can be kept "clean" - if this PR makes it easier to accidentally include uncommitted custom code by changing one source tree and including those changes in the other package, then that's not good.

I think that the current mechanism of cleaning the working tree at the beginning of a build should be kept. The risk here would be of the user losing custom code changes (i.e., if he checks in a source using `conda build --source`, and then modify this source). I don't view that as a problem, as long as it is well documented.

**Comment by pelson** [comment]
Just as a heads up - there are two directories which are important when building:

 * the source to compile, this is the "work" directory
 * the target prefix, this is the "_build" environment

Both of these would need to be changed to be able to build in parallel. FWIW, I've got some code which produces temporary conda environments which are automatically pruned after 25 hrs (well, as soon as the next process fires up anyway) - that would basically solve the "_build" directory side of things. Sounds like you are looking at the "work" directory side of things.

**Comment by msarahan** [comment]
This is implemented in https://github.com/conda/conda-build/pull/2249, closing.
